### PR TITLE
Template versioning support

### LINKED ISSUE #957: Release first version of CLI to Maven central
Just for me not to forget it, I should:

- [x] Change `mmm` dependencies to latest released ones.
- [x] Check if any code needs to be changed in order to use them.
- [ ] Add `mmm` dependencies and `core` to artificial pom so that the CobiGen.jar is very lightweight.
- [ ] Release CobiGen CLI to Maven central-

### PR DESCRIPTION
**Pull request by Mghcapgemini**
Adresses/Fixes #957 .

Implements

* Template versioning support

@devonfw/cobigen

### CODE REVIEW

File: cobigen-cli/pom.xml [hunk 1]
```diff
@@ -24,7 +24,7 @@
     <dependency>
       <groupId>com.devonfw.cobigen</groupId>
       <artifactId>core-api</artifactId>
-      <version>5.3.2</version>
+      <version>5.4.0-SNAPSHOT</version>
```
Comment (jdiazgon) [inline cobigen-cli/pom.xml]: The last version of the core that we should use is `5.3.4`

File: cobigen-cli/src/main/java/com/devonfw/cobigen/cli/utils/CobiGenUtils.java [hunk 2]
```diff
@@ -352,12 +362,29 @@ public void addJarsToClassLoader(String allJars) {
...
+            }
+        }
         if (TemplatesJarUtil.getJarFile(isSource, jarFileDir) == null) {
             try {
-                TemplatesJarUtil.downloadLatestDevon4jTemplates(isSource, jarFileDir);
+                if (!templateDependencyIsGiven()) {
+                    TemplatesJarUtil.downloadLatestDevon4jTemplates(isSource, jarFileDir);
+                } else {
+                    Dependency templatesDep = getTemplateDependency();
+                    TemplatesJarUtil.downloadJar(templatesDep.getGroupId(), templatesDep.getArtifactId(),
```
[the remark this answers is not in the document]
Comment (jdiazgon) [reply cobigen-cli/src/main/java/com/devonfw/cobigen/cli/utils/CobiGenUtils.java]: Also I have found out that with the current implementation, it creates a `null` folder next to the input file containing `null\com\devonfw\cobigen\templates-devon4j\3.1.6\templates-devon4j-3.1.6.jar`.

image

File: cobigen-cli/src/main/resources/pom.xml [hunk 3]
```diff
@@ -63,7 +63,12 @@
         <artifactId>jsonplugin</artifactId>
         <version>2.0.0</version>
       </dependency>
-          <dependency>
+      <dependency>
+        <groupId>com.devonfw.cobigen</groupId>
+        <artifactId>templates-devon4j</artifactId>
+        <version>3.1.6</version>
+      </dependency>
+    <dependency>
```
Comment (jdiazgon) [inline cobigen-cli/src/main/resources/pom.xml]: Can you change the `tsPlugin` version to `2.3.1`? I have just released it today.

File: cobigen-cli/src/main/java/com/devonfw/cobigen/cli/utils/CobiGenUtils.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (Mghcapgemini) [reply cobigen-cli/src/main/java/com/devonfw/cobigen/cli/utils/CobiGenUtils.java]: Folder creation is now deactivated.

**Pull request merged**